pkg/process/monitor/process_monitor_test.go

```diff
@@ -34,19 +34,61 @@ func getProcessMonitor(t *testing.T) *ProcessMonitor {
 	return pm
 }
 
+// pidRecorder is a helper to record pids and check if they were recorded.
+type pidRecorder struct {
+	mu   sync.Mutex
+	pids map[uint32]struct{}
+}
+
+// newPidRecorder creates a new pidRecorder.
+func newPidRecorder() *pidRecorder {
+	return &pidRecorder{pids: make(map[uint32]struct{})}
+}
+
+// record records a pid.
+func (pr *pidRecorder) record(pid uint32) {
+	pr.mu.Lock()
+	defer pr.mu.Unlock()
+	pr.pids[pid] = struct{}{}
+}
+
+// has checks if a pid was recorded.
+func (pr *pidRecorder) has(pid uint32) bool {
+	pr.mu.Lock()
+	defer pr.mu.Unlock()
+	_, ok := pr.pids[pid]
+	return ok
+}
+
+// getProcessCallback returns a ProcessCallback wrapper of the pidRecorder.record method.
+func getProcessCallback(r *pidRecorder) *ProcessCallback {
+	f := func(pid uint32) {
+		r.record(pid)
+	}
+	return &f
+}
+
+func getProcessCallback2(t *testing.T, r *pidRecorder, typ string) *ProcessCallback {
```

Comment:
is this necessary?

Reply:
nope, the PR was in a draft mode and it was part of debugging the failures in the CI